cmd/nodecmd/create_aws.go

```diff
@@ -139,41 +139,54 @@ func createEC2Instances(ec2Svc map[string]*awsAPI.AwsCloud,
 			return nil, nil, nil, nil, err
 		}
 		certInSSHDir, err := app.CheckCertInSSHDir(regionConf[region].CertName)
+		if useSSHAgent {
+			certInSSHDir = true // if using ssh agent, we consider that we have a cert on hand
+		}
 		if err != nil {
 			return nil, nil, nil, nil, err
 		}
-		certName := regionConf[region].CertName
+		sgID := ""
 		keyPairName[region] = regionConf[region].Prefix
 		securityGroupName := regionConf[region].SecurityGroupName
-		privKey, err := app.GetSSHCertFilePath(certName)
-		sgID := ""
+		privKey, err := app.GetSSHCertFilePath(regionConf[region].CertName)
```

Comment:
maybe for readability we need two function to replace here, one for the ssh agent case, and the other for the ssh file case

Reply:
not sure wdym. I suggest leaving this as is unless you want to open a PR with your implementation

Reply:
i think you want to fix this massibe if/switch statement combo.
we should do it, but maybe in separate PR